javascript/lib.c

```diff
@@ -89,8 +90,8 @@ napi_value CountSubstrAPI(napi_env env, napi_callback_info info) {
 
     size_t result = 0;
 
-    if (haystack_l == 1)
-        result = count_char(strzl_haystack, needle);
+    if (needle_l == 1 || needle_l == 0)
```

Comment:
In case the pattern is an empty string we can exit early. No need to call the counter.